### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by Anis1988**

### CODE REVIEW

**Review by TrisTenuto — APPROVED** [review]

File: Repository/Models/Cinephiliacs_ReviewContext.cs [hunk 1]
```diff
@@ -24,7 +24,8 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
         {
             if (!optionsBuilder.IsConfigured)
             {
-                optionsBuilder.UseSqlServer("Server=.\\SQLExpress;Database=ReviewDB;Trusted_Connection=True;");
+#warning To protect potentially sensitive information in your connection string, you should move it out of source code. You can avoid scaffolding the connection string by using the Name= syntax to read it from configuration - see https://go.microsoft.com/fwlink/?linkid=2131148. For more guidance on storing connection strings, see http://go.microsoft.com/fwlink/?LinkId=723263.
```
Comment (markmoorerev) [inline Repository/Models/Cinephiliacs_ReviewContext.cs:27]: is there a plan to move this?

File: ReviewApi/Controllers/ReviewController.cs [hunk 2]
```diff
@@ -104,35 +109,95 @@ public async Task<ActionResult> SetReviewsPageSize(int pagesize)
...
+            if(await _reviewLogic.CreateReview(reviewDto))
             {
                 return StatusCode(201);
             }
-            else
+            
+            return StatusCode(400);
+            
+        }
+        [HttpPatch("update/{reviewId}")]
```
Comment (markmoorerev) [inline ReviewApi/Controllers/ReviewController.cs:153]: comments

File: ReviewApi/Controllers/ReviewController.cs [hunk 3]
```diff
@@ -104,35 +109,95 @@ public async Task<ActionResult> SetReviewsPageSize(int pagesize)
...
+                reviewDto.ReviewId = reviewExist.ReviewId;
+                _reviewLogic.UpdatedRev(reviewDto);
+                return  StatusCode(200);
+            }
+
+            return  StatusCode(404);
+
+        }
+
+        [HttpDelete("delete/{reviewId}")]
```
Comment (markmoorerev) [inline ReviewApi/Controllers/ReviewController.cs:174]: comments

File: ReviewApi/Controllers/ReviewController.cs [hunk 4]
```diff
@@ -104,35 +109,95 @@ public async Task<ActionResult> SetReviewsPageSize(int pagesize)
...
+            if (rev != null)
+            {
+                _reviewLogic.deleteReview(rev);
+                return  StatusCode(200);
+            }
+
+            return StatusCode(404);
+        }
+
+        [HttpGet("/byRating/{rating}")]
```
Comment (markmoorerev) [inline ReviewApi/Controllers/ReviewController.cs:191]: c

File: ReviewApi/Controllers/ReviewController.cs [hunk 5]
```diff
@@ -104,35 +109,95 @@ public async Task<ActionResult> SetReviewsPageSize(int pagesize)
...
+            }
+            if(reviews.Count == 0)
+            {
+                return StatusCode(204);
+            }
+            StatusCode(200);
+            return reviews;
+        }
+        
+        [HttpGet("/byRating/{imdb}/{rating}")]
```
Comment (markmoorerev) [inline ReviewApi/Controllers/ReviewController.cs:213]: o

**Review by markmoorerev — CHANGES_REQUESTED** [review]
mostly just lacking comments

**Pull request closed without merging**